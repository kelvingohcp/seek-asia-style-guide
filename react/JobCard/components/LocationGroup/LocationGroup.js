import React from 'react';
import PropTypes from 'prop-types';
import styles from './LocationGroup.less';
import defaultLink from '../Link/Link';

export const LocationsPropTypes = PropTypes.arrayOf(PropTypes.shape({
  name: PropTypes.string,
  link: PropTypes.string,
  title: PropTypes.string,
  child: PropTypes.shape({
    name: PropTypes.string,
    link: PropTypes.string,
    title: PropTypes.string,
    child: PropTypes.shape({
      name: PropTypes.string
    })
  })
}));

const renderLocation = ({ link, name, title, child, LinkComponent, key }) => {
  const locationLink = (link) ?
    (<LinkComponent link={link} className={styles.locationLink} key={key} title={title}>{name}</LinkComponent>) :
    (<span className={styles.locationName} key={key}>{name}</span>);
  if (child) {
    return [locationLink, ' > ', ...renderLocation({ ...child, LinkComponent, key: `${key}1` })];
  }
  return [locationLink];
};

const LocationGroup = ({ locations, LinkComponent = defaultLink }) => {
  return locations.map((location, index) =>
    renderLocation({ ...location, LinkComponent, key: index })
  ).reduce((prev, curr) => [prev, ', ', ...curr]);
};

export default LocationGroup;

LocationGroup.propTypes = {
  locations: LocationsPropTypes.isRequired,
  LinkComponent: PropTypes.func
};
