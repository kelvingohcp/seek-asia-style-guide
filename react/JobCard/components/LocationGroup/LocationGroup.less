@import (reference) "~seek-asia-style-guide/theme";
@import (reference) '../../JobCard.less';

.locationName {
  &:extend(.jobInfo);
}

.locationLink {
  &:extend(.jobInfo);
  .link(@saGrey1, true);
}
