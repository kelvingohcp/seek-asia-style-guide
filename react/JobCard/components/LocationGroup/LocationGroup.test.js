import React from 'react';
import { shallow } from 'enzyme';

import LocationGroup from './LocationGroup';

const defaultLocations = [
  {
    name: 'Pahang',
    link: '/jobCard'
  },
  {
    name: 'Selangor',
    link: '/jobCard',
    child: {
      name: 'Cheras',
      link: '/jobCard',
      child: {
        name: 'Near Leisure Mall'
      }
    }
  }
];

describe('JobCard - LocationGroup', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<LocationGroup locations={defaultLocations} />);
    expect(wrapper).toMatchSnapshot();
  });
});
