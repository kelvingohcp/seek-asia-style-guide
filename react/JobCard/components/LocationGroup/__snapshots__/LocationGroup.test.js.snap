// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCard - LocationGroup should render correctly 1`] = `
Array [
  Array [
    <Link
      className="locationLink"
      key="0"
      link="/jobCard"
      title="Limit result to Pahang"
    />,
  ],
  ", ",
  <Link
    className="locationLink"
    key="1"
    link="/jobCard"
    title="Limit result to Selangor"
  />,
  " > ",
  <Link
    className="locationLink"
    key="11"
    link="/jobCard"
    title="Limit result to Cheras in Selangor"
  />,
  " > ",
  <span
    className="locationName"
    key="111"
  />,
]
`;
